### PR TITLE
Add example cmake project

### PR DESCRIPTION
**Pull request by jrhemstad**
## Description

fixes https://github.com/NVIDIA/cccl/issues/131

This provides a simple, standalone project that demonstrates how to use the CCCL monorepo from a cmake project. 

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: examples/example_project/CMakeLists.txt
Comment (jrhemstad) [inline examples/example_project/CMakeLists.txt:23]: @allisonvacanti this isn't currently actually possible, but I wanted to talk to you about if/how we could make this work.

The gist is, what target do we want to tell most people to link against to get all 3 libraries?

File: examples/example_project/CMakeLists.txt
Comment (jrhemstad) [inline examples/example_project/CMakeLists.txt:14]: I think we should default these options to `OFF`. We should optimize for making it easy for external users to use our cmake, and they don't care about enabling our tests. They just want our targets to link against.

**Comment by jrhemstad** [comment]
This is currently blocked by https://github.com/NVIDIA/cccl/issues/203

**Comment by jrhemstad** [comment]
I realized it made more sense to have CPM point to this PR branch for now, so I updated that (still doesn't work).

File: examples/CMakeLists.txt
Comment (allisonvacanti) [inline examples/CMakeLists.txt:1]: I think the intention here is to allow users to run all of the examples at once by configuring this directory?

I think this is unnecessary -- Adding a `CCCL_ENABLE_EXAMPLES` option to the top level CCCL CMakeLists.txt that conditionally does `add_subdirectory(examples)` will let us check that the examples are working in CI, and if users want to try out a single example, they can just configure those directly (without even configuring CCCL), which would be more illustrative as an example than configuring this multi-example harness.

**Comment by jrhemstad** [comment]
Remember to add license headers

File: examples/example_project/example.cu
Comment (jrhemstad) [inline examples/example_project/example.cu:19]: Don't use magic 256 for block size. Use a constexpr global or a template param.

File: examples/example_project/CMakeLists.txt
Comment (allisonvacanti) [inline examples/example_project/CMakeLists.txt:12]: Make sure to update this before merging.

**Review by allisonvacanti — APPROVED** [review]

File: examples/example_project/CMakeLists.txt
Comment (allisonvacanti) [reply examples/example_project/CMakeLists.txt]: No -- the same process of CMake never executes both the main CCCL build *and* the example project. They're totally isolated.

We could change the comment and just say that we're giving an example of adding a regression test in CMake and hide our hacks in plain sight...

File: examples/example_project/example.cu
Comment (senior-zero) [inline examples/example_project/example.cu:48]: `num_blocks, block_size` or `numBlocks, blockSize`.

**Review by senior-zero — APPROVED** [review]

File: CMakePresets.json
Comment (allisonvacanti) [reply CMakePresets.json:28]: Yep -- this can be overridden by either passing `-DCMAKE_CUDA_ARCHITECTURES` to cmake (in addition to `--preset`), or modifying the config with ccmake/cmake-gui.

I only added it because the example uses libcu++ atomics, so the default arch (5.2) will throw errors for being too old. If the example avoided atomics we could drop this completely.

**Pull request merged**
